crypto/keys/keybase.go

```diff
@@ -0,0 +1,364 @@
+		priv, err = crypto.NewPrivKeyLedgerSecp256k1(linfo.Path)
+		if err != nil {
+			return
+		}
+	case offlineInfo:
+		linfo := info.(offlineInfo)
+		fmt.Printf("Bytes to sign:\n%s", msg)
+		buf := bufio.NewReader(os.Stdin)
+		fmt.Printf("\nEnter Amino-encoded signature:\n")
+		// Will block until user inputs the signature
+		signed, err := buf.ReadString('\n')
+		if err != nil {
+			return nil, nil, err
+		}
+		cdc.MustUnmarshalBinary([]byte(signed), sig)
+		return sig, linfo.GetPubKey(), nil
+	}
+	sig, err = priv.Sign(msg)
+	if err != nil {
+		return nil, nil, err
+	}
+	pub = priv.PubKey()
+	return sig, pub, nil
+}
+
+func (kb dbKeybase) Export(name string) (armor string, err error) {
+	bz := kb.db.Get(infoKey(name))
+	if bz == nil {
+		return "", fmt.Errorf("no key to export with name %s", name)
+	}
+	return armorInfoBytes(bz), nil
+}
+
+// ExportPubKey returns public keys in ASCII armored format.
+// Retrieve a Info object by its name and return the public key in
+// a portable format.
+func (kb dbKeybase) ExportPubKey(name string) (armor string, err error) {
+	bz := kb.db.Get(infoKey(name))
+	if bz == nil {
+		return "", fmt.Errorf("no key to export with name %s", name)
+	}
+	info, err := readInfo(bz)
+	if err != nil {
+		return
+	}
+	return armorPubKeyBytes(info.GetPubKey().Bytes()), nil
+}
+
+func (kb dbKeybase) Import(name string, armor string) (err error) {
+	bz := kb.db.Get(infoKey(name))
+	if len(bz) > 0 {
+		return errors.New("Cannot overwrite data for name " + name)
+	}
+	infoBytes, err := unarmorInfoBytes(armor)
+	if err != nil {
+		return
+	}
+	kb.db.Set(infoKey(name), infoBytes)
+	return nil
+}
+
+// ImportPubKey imports ASCII-armored public keys.
+// Store a new Info object holding a public key only, i.e. it will
+// not be possible to sign with it as it lacks the secret key.
+func (kb dbKeybase) ImportPubKey(name string, armor string) (err error) {
+	bz := kb.db.Get(infoKey(name))
+	if len(bz) > 0 {
+		return errors.New("Cannot overwrite data for name " + name)
+	}
+	pubBytes, err := unarmorPubKeyBytes(armor)
+	if err != nil {
+		return
+	}
+	pubKey, err := tcrypto.PubKeyFromBytes(pubBytes)
+	if err != nil {
+		return
+	}
+	kb.writeOfflineKey(pubKey, name)
+	return
+}
+
+// Delete removes key forever, but we must present the
+// proper passphrase before deleting it (for security).
+// A passphrase of 'yes' is used to delete stored
+// references to offline and Ledger / HW wallet keys
+func (kb dbKeybase) Delete(name, passphrase string) error {
+	// verify we have the proper password before deleting
+	info, err := kb.Get(name)
+	if err != nil {
+		return err
+	}
+	switch info.(type) {
+	case localInfo:
+		linfo := info.(localInfo)
+		_, err = unarmorDecryptPrivKey(linfo.PrivKeyArmor, passphrase)
+		if err != nil {
+			return err
+		}
+		kb.db.DeleteSync(infoKey(name))
+		return nil
+	case ledgerInfo:
+	case offlineInfo:
+		if passphrase != "yes" {
+			return fmt.Errorf("enter exactly 'yes' to delete the key")
```

Comment:
I think should be changed to `enter 'yes' exactly`. Perhaps indicate that this can't be undone, unless that was indicated elsewhere.

Reply:
Done.